sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy

```diff
@@ -96,222 +94,345 @@ class APISpec extends Specification {
 
     static final String garbageLeaseID = UUID.randomUUID().toString()
 
-    /*
-    credential for various kinds of accounts.
-     */
-    @Shared
-    static SharedKeyCredential primaryCreds
+    static def AZURE_TEST_MODE = "AZURE_TEST_MODE"
+    static def PRIMARY_STORAGE = "PRIMARY_STORAGE_"
+    static def SECONDARY_STORAGE = "SECONDARY_STORAGE_"
+    static def BLOB_STORAGE = "BLOB_STORAGE_"
+    static def PREMIUM_STORAGE = "PREMIUM_STORAGE_"
 
-    @Shared
-    static SharedKeyCredential alternateCreds
+    static SharedKeyCredential primaryCredential
+    static SharedKeyCredential alternateCredential
+    static SharedKeyCredential blobCredential
+    static SharedKeyCredential premiumCredential
+    static TestMode testMode
 
-    /*
-    URLs to various kinds of accounts.
-     */
-    BlobServiceClient primaryServiceURL
+    BlobServiceClient primaryServiceClient
```

Comment:
I'd suggest we use the same way to define the variable. Xstore team suggests to use **def** as much as possible.

Reply:
I'd prefer not to use `def` here as it cannot infer type as it is assigned during runtime and it makes it much harder to program as IntelliSense is lost.